Dockerfile

```diff
@@ -0,0 +1,26 @@
+# Base
+FROM python as base
+
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+WORKDIR /code
+
+ENV PATH "/code:/root:/root/.poetry/bin:${PATH}"
+
+# add and install python requirements
+COPY pyproject.toml poetry.lock /code/
+RUN poetry install
```

Comment:
Note that you don't actually need a virtual env inside the container. 
And also the production image doesn't need e.g. pytest.

So you could do
```
RUN poetry config virtualenvs.create false --local && poetry install --no-dev --no-root
```